tsdb/wal/wal.go

```diff
@@ -774,6 +784,52 @@ func listSegments(dir string) (refs []segmentRef, err error) {
 	return refs, nil
 }
 
+// Get segments from the directory
+func getSegments(dir string) ([]segmentRef, error) {
+	files, err := fileutil.ReadDir(dir)
+	if err != nil {
+		return nil, err
+	}
+	var refs []segmentRef
+	for _, fn := range files {
+		k, err := strconv.Atoi(fn)
+		if err != nil {
+			return nil, err
+		}
+		refs = append(refs, segmentRef{name: fn, index: k})
+	}
+	sort.Slice(refs, func(i, j int) bool {
+		return refs[i].index < refs[j].index
+	})
+	return refs, nil
+}
```

Comment:
Can we avoid duplication of code? `listSegments` could call `getSegments` and check for the sequence.